kmath-trajectory/src/commonMain/kotlin/space/kscience/kmath/trajectory/route.kt

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2018-2021 KMath contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package space.kscience.kmath.trajectory
+
+public fun interface MaxCurvature {
+    public fun compute(startPoint: PhaseVector2D): Double
+}
+
+public fun DubinsPath.Companion.shortest(
+    start: PhaseVector2D,
+    end: PhaseVector2D,
+    computer: MaxCurvature,
+): DubinsPath = shortest(start, end, computer.compute(start))
```

Comment:
I'm not sure if this interface is good. I would rather say that the computer should take a segment and behaviour for this segment. The length is calculated differently for the line, the arc, or the trochoid. We can estimate the time or energy consumption. The space of velocity can change from segment to segment. In general, the speed may vary in one segment. But it can be disI'm not sure if this interface is good. I would rather say that the computer should take a segment and behaviour for this segment. The length is calculated differently for the line, the arc, or the trochoid. We can estimate the time or energy consumption. The space of velocity can change from segment to segment. In general, the speed may vary in one segment. But it can be discussed later.

Reply:
Indeed, I did not find a good solution yet for a a path with changed curvature. As for estimator, it will be added later on top of it. I can't imaging the case where the shortest path is not the best yet.